nightingale_dispatcher/src/nightingale_dispatcher/mission_planner.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+import rospy
+import actionlib
+import queue
+
+from actionlib_msgs.msg import GoalStatus
+from nightingale_msgs.msg import MissionPlanAction
+from nightingale_dispatcher.handoff_task import HandoffTask
+from nightingale_dispatcher.idle_task import IdleTask
+from nightingale_dispatcher.navigate_task import NavigateTask
+from nightingale_dispatcher.stock_task import StockTask
+from nightingale_dispatcher.task import Task
+from nightingale_dispatcher.triage_task import TriageTask
+
+class MissionPlanner:
+    def __init__(self):
+        rospy.init_node("mission_planner_node")
+
+        self.server = actionlib.SimpleActionServer(
+            "mission_planner", MissionPlanAction, self.goal_cb, False
+        )
+        self.server.start()
+
+        self.handoff_task = HandoffTask()
+        self.idle_task = IdleTask()
+        self.navigate_task = NavigateTask()
+        self.stock_task = StockTask()
+        self.triage_task = TriageTask()
+
+        self.states = queue.Queue()
+    
+    def go_to_patient(self):
```

Comment:
I think we also need a function to go back to the home position. It seems like the idle function should be the very last thing called before the mission planner exits and waits for the next service request

Reply:
What is the expected behaviour between the proposed go_home and go_idle?

Reply:
I was thinking the go_home task brings the robot home from whereever it is and then queues up the go_idle task to exit. If you were already thinking of this for the idle task then this isn't necessary

Reply:
I was thinking that would go in the go_idle task, ie go_idle drives to the idle position then waits for a new task

Reply:
actually i dont think that can go in IdleTask. IdleTask doesnt have access to navigate_task so it cant call execteu. i can add this in my PR

Reply:
The IdleTask can also call room runner. Essentially for the IdleTask, the only thing to do is call room runner with the idle home room and pose